src/components/sidenav/sidenav.spec.ts

```diff
@@ -0,0 +1,286 @@
+import {
+  it,
+  iit,
+  describe,
+  ddescribe,
+  expect,
+  inject,
+  injectAsync,
+  TestComponentBuilder,
+  beforeEachProviders,
+  beforeEach,
+} from 'angular2/testing';
+import {provide, Component, DebugElement} from 'angular2/core';
+import {By} from 'angular2/platform/browser';
+
+import {MdSidenav, MdSidenavLayout, MD_SIDENAV_DIRECTIVES} from './sidenav';
+import {AsyncTestFn, FunctionWithParamTokens} from 'angular2/testing';
+import {ComponentFixture} from "angular2/testing";
+import {EventEmitter} from "angular2/core";
+import {Predicate} from "angular2/src/facade/collection";
+
+
+function wait(msec: number) {
+  return new Promise<void>(resolve => window.setTimeout(() => resolve(), msec));
+}
+
+
+function waitOnEvent(fixture: ComponentFixture,
+                     by: Predicate<DebugElement>,
+                     propertyName: string) {
+  fixture.detectChanges();
+
+  // Wait for the animation end.
+  return new Promise(resolve => {
+    const component: any = fixture.debugElement.query(by).componentInstance;
+    component[propertyName].subscribe(resolve);
+  });
+}
+
+
+describe('MdSidenav', () => {
+  let builder: TestComponentBuilder;
+
+  beforeEach(inject([TestComponentBuilder], (tcb: TestComponentBuilder) => { builder = tcb; }));
+
+  describe('methods', () => {
+    it('should be able to open and close', (done: any) => {
+      let testComponent: BasicTestApp;
+      let fixture: ComponentFixture;
```

Comment:
What do you gain from declaring fixture out here? Why not just do `.then(fixture: ComponentFixture)` in each of the tests and save some lines?